File: Using Java/153. Find Minimum in Rotated Sorted Array.java
public class Solution {
    public int findMin(int[] nums) {
        int s = 0;
        int end = nums.length - 1;
        
        while (s < end) {
            int mid = (s + end) / 2;
            
            if (nums[mid] > nums[end]) {
                s = mid + 1;
            } else {
                end--;
            }
        }
        
        return nums[s];
    }
}

File: using python/153. Find Minimum in Rotated Sorted Array.py
class Solution:
    def findMin(self, nums):
        s, end = 0, len(nums)-1
        while s < end:
            mid = (s + end) // 2         
            if nums[mid] > nums[end]:
                s = mid + 1
            else:
                end -=1 
                
        return(nums[s])
